lib/game_collection.rb

```diff
@@ -30,4 +30,8 @@ def total_goals_per_game
   def all
     @games
   end
+
+  def array_by_key(key)
+    @games.map{|game| game.send "#{key}" }.uniq  ## can probably put this in a module passing class, collection, and key as arguments
```

Comment:
Nice work on using send! I haven't seen this syntax before, but I trust this works since your tests are passing